### ISSUE TITLE
How to set a separate theme for Vifm.vim?

### ISSUE THREAD

**Issue by mohabme** [vifm/vifm.vim#72]
Even without using `:VifmCs`, when launched through the plugin's commands, _Vifm.vim_ always uses _Vim_'s theme, or an approximation thereof, and _Vifm_'s native `:colorscheme` command doesn't have any effect.

Can this behavior be disabled/altered so I can set a separate theme for _Vifm.vim?_

**Comment by xaizek** [comment]
> Vifm.vim always uses Vim's theme

It can't, it probably just looks like it.

Please provide terminal information from the bottom of `:vifm` menu when Vifm is launched inside Vim, it might be helpful. I suspect `$TERM` value gets set to something that supports only 16 colors, this is especially likely when using gVim.

**Comment by mohabme** [comment]
>Please provide terminal information from the bottom of `:vifm` menu when Vifm is launched inside Vim, it might be helpful.

![2021-12-15-022321_400x177_scrot](https://user-images.githubusercontent.com/96145196/146102651-5d83cc47-2fc0-4941-86a8-9c614d0eb9cd.png)

`echo $TERM` from inside _Vifm.vim_ returns `xterm-256color` too. I'm using terminal-based _Neovim_, and _xfce4-terminal_.

**Comment by mohabme** [comment]
https://user-images.githubusercontent.com/96145196/146198979-9b47dee4-bdc6-4ca2-8a0e-51e6b4b35422.mp4

This is how the plugin behaves; I think only the background color tries to match that from _Vim_'s current color scheme. 

In addition to that, `:colorscheme` does not work inside _Vifm.vim_. It doesn't throw an error, but it doesn't have any effect.

**Comment by xaizek** [comment]
Oh, that's just "default" colors. Terminal has one for background and foreground and programs can use them to match terminal's theme, builtin terminals have them too. You can modify Vifm's color scheme to not have "default", "-1" or "None" for colors to get rid of this effect.

`:colorscheme` should work though unless Neovim's terminal is doing something with the palette. Vifm just uses colors as they are specified in the color scheme and terminal information looks correct. Having
```vim
highlight Win   ctermbg=default
```
in colorschemes is quite common, so I wouldn't be surprised if it's responsible for an impression that colorscheme isn't changed.

**Comment by mohabme** [comment]
Thank you for the helpful information, I think `:colorscheme` worked all along, but the background color matching _Neovim_'s color scheme change is what confused me. I'm beginning to figure it out, but shouldn't both _Vifm.vim_ and _Vifm_ show the same color if I'm using the same color scheme?

This is _Vifm.vim_, launched from inside _Neovim_ with `:TabVifm` command, and displaying the output of `:colorscheme?` at the bottom:
![2021-12-15-170344_1364x766_scrot](https://user-images.githubusercontent.com/96145196/146213167-1051132f-733e-43d0-92ff-7eb8043e9807.png)

This is _Vifm_ launched directly from _xfce4-terminal_:
![2021-12-15-170358_1364x766_scrot](https://user-images.githubusercontent.com/96145196/146213340-bda008c5-237e-42c6-a60e-a2e3161b0ff1.png)

Some options seem to be defaulting to terminal colors in _Vifm.vim_, but not in _Vifm_ even though I'm using the same color scheme for both. They're both loading form the same file, essentially, but displaying different results.

**Comment by xaizek** [comment]
Things like `nmap`, `vmap` highlighted in red suggest that Neovim applies syntax rules to the terminal buffer, see if `'syntax'` is set for it. I think Vim automatically disables things that make no sense in terminal buffers, but Neovim might need to have them disabled explicitly.

**Comment by mohabme** [comment]
I tried `:syntax clear`, `:syntax off`, and `:ownsyntax off`, but the problem presists; however, `:set nospell` did the trick. 

I tried `:syntax spell notoplevel` but it wasn't enough to disable highlighting spelling errors inside _Vifm.vim_; however, I can do something like `:setlocal nospell | :Vifm` to get _Vifm.vim_ to launch with a clean color scheme, and `:tabnew | :setlocal nospell | :Vifm` as an alternative to `:TabVifm`.

It seems the behavior is tied to _Neovim_ spell-checking comments, so as long as that is enabled, it will continue to check `:Vifm` output for spelling mistakes whenever it's launched with any of the plugin's commands. Unless there's a _Neovim_ option I missed that can disable spell-checking/highlighting spelling mistakes in this particular scenario...

Thank you for your help. If there's nothing else that can be done, I think I'm satisifed with this solution for the most part, so I will close this issue tomorrow.
